File: packages/platform-ios/Sources/Keyward/KeyRegistry.swift
public class KeyRegistry {
    private var currentUserId: String?

    public init() {}

    public func setUserId(_ userId: String) {
        currentUserId = userId
    }

    public func clearUserId() {
        currentUserId = nil
    }

    public func getUserId() -> String? {
        currentUserId
    }

    public func resolve(_ keyDef: KeyDef) throws -> String {
        switch keyDef.scope {
        case .user:
            guard let userId = currentUserId else {
                throw KeywardError.userIdNotSet
            }
            return "u/\(userId)/\(keyDef.key)"
        case .device:
            return "d/\(keyDef.key)"
        case .global:
            return keyDef.key
        }
    }

    public static func getUserPrefix(_ userId: String) -> String {
        "u/\(userId)/"
    }

    public func getUserPrefix(_ userId: String) -> String {
        Self.getUserPrefix(userId)
    }
}

File: packages/platform-android/src/main/java/org/keyward/KeyRegistry.java
package org.keyward;

public final class KeyRegistry {
    private String currentUserId;

    public KeyRegistry() {}

    public void setUserId(String userId) {
        this.currentUserId = userId;
    }

    public void clearUserId() {
        this.currentUserId = null;
    }

    public String getUserId() {
        return currentUserId;
    }

    public String resolve(KeyDef keyDef) throws KeywardException {
        switch (keyDef.getScope()) {
            case USER:
                if (currentUserId == null) {
                    throw KeywardException.userIdNotSet();
                }
                return "u/" + currentUserId + "/" + keyDef.getKey();
            case DEVICE:
                return "d/" + keyDef.getKey();
            case GLOBAL:
                return keyDef.getKey();
            default:
                throw new IllegalStateException("Unknown scope: " + keyDef.getScope());
        }
    }

    public static String getUserPrefix(String userId) {
        return "u/" + userId + "/";
    }
}
